fil_fungible_token/src/token/mod.rs

```diff
@@ -49,6 +50,53 @@ pub enum TokenError {
     Serialization(#[from] SerializationError),
 }
 
+impl From<TokenError> for ExitCode {
+    fn from(error: TokenError) -> Self {
+        match error {
+            TokenError::ReceiverHook { from: _, to: _, operator: _, amount: _, exit_code } => {
+                exit_code
+            }
+            TokenError::InvalidIdAddress { address: _, source: _ } => ExitCode::USR_NOT_FOUND,
+            TokenError::Serialization(_) => ExitCode::USR_SERIALIZATION,
+            TokenError::InvalidNegative(_) => ExitCode::USR_ILLEGAL_ARGUMENT,
+            TokenError::State(state_error) => match state_error {
+                StateError::IpldHamt(_) | StateError::Serialization(_) => {
+                    ExitCode::USR_SERIALIZATION
+                }
+                StateError::MissingState(_) => ExitCode::USR_ILLEGAL_STATE,
+                StateError::NegativeBalance { balance: _, delta: _, owner: _ }
+                | StateError::InsufficentAllowance {
+                    owner: _,
+                    operator: _,
+                    allowance: _,
+                    delta: _,
+                } => ExitCode::USR_FORBIDDEN,
+                StateError::NegativeTotalSupply { supply: _, delta: _ } => ExitCode::USR_FORBIDDEN,
```

Comment:
Where does this come from? It seems like an internal invariant failure, in which case ILLEGAL_STATE or ASSERTION_FAILED